internal/core/component.rs

```diff
@@ -111,6 +111,18 @@ pub struct ComponentVTable {
         result: &mut SharedString,
     ),
 
+    /// Returns a Window (if one is set up yet!)
+    pub maybe_window_adapter: extern "C" fn(
+        core::pin::Pin<VRef<ComponentVTable>>,
+        result: &mut Option<Rc<dyn WindowAdapter>>,
+    ),
+
+    /// Returns a Window, creating a fresh one if necessary
+    pub window_adapter: extern "C" fn(
+        core::pin::Pin<VRef<ComponentVTable>>,
+        result: &mut Option<Rc<dyn WindowAdapter>>,
```

Comment:
Did you not mention this should return an` &mut Result<Rc<dyn WindowAdapter>>, PlatformError>` ?
Also, instead of two functions, did you consider taking a `bool: create` ?